perlguts: Add a paragraph about newSVpvz()

khwilliamson opened the pull request.

leonerd commented on pod/perlguts.pod:
```diff
@@ -117,6 +117,11 @@ the SV has the undef value.
 
+Or you can create an SV that is initialized to an empty string, by using
+L<C<newSVpvz(len)>|perlapi/newSVpvz>.  Perl will not have to ask for
+more space for it from the operating system until and unless the string
```
I'm not sure if we need to be this specific about "from the operating system", unless there's something more specific going on here. Maybe just some words about "Perl will not have to reallocate space unless and until the string ..."

khwilliamson committed:
perlguts: Update info about creating and setting SV

New SV functions have been added since this was last updated.  Change
the wording to also point where the definitive current list is located.

khwilliamson committed:
perlguts: Add a paragraph about newSVpvz

khwilliamson committed:
perlapi: Distinguish between newSVpvz and newSV

tonycoz commented on pod/perlguts.pod:
```diff
@@ -90,24 +90,34 @@ is documented only in this paragraph.)
 F<config.h>) guaranteed to be large enough to represent the size of
 any string that perl can handle.
 
+S<C<"literal string">> is a string enclosed in double quote characters,
+and the created SV is of string type (PV), initialized to contain the
+literal string parameter. C<newSVpvs("")> creates an SV containing an
```
This could be read as SvPVX pointing at the supplied literal, maybe:
> initialized to contain a copy of the literal string parameter.
